src/olympia/addons/utils.py

```diff
@@ -17,19 +17,43 @@ def generate_addon_guid():
 
 
 def reverse_name_lookup(key, addon_type):
+    """Does a reverse lookup by localized add-on name.
+
+    If `key` is a dictionary we assume it's a localized mapping of
+
+    .. code-block::
+
+        {'en-us': 'Bands Make You Dance'}
+
+    :returns: a tuple of `(id, list_of_locales)` while `list_of_locales`
+              defaults to the add-on default locale if no localization is
+              used. `None` is being returned if there are no results.
```

Comment:
The docstring appears to be out of date, since you no longer return the list of locales, right ?

I feel like this function could be improved further: if you pass it the add-on id, (or `None` in the case of a new add-on), then you can just exclude it in the queryset, and just return `qs.exists()` instead of doing a `.values_list()`. That would also simplify the `clean_addon_name()` method in the form above. Maybe worth doing in a separate bug though?